src/components/Table/index.tsx

```diff
@@ -0,0 +1,219 @@
+  flexDirection: 'column',
+  position: 'absolute',
+  right: '24px',
+  top: `${props.itemIndex * 50 + 58}px`,
+  background: '#FFFFFF',
+  boxShadow: '0px 4px 12px rgba(0, 6, 41, 0.1)',
+  zIndex: 200,
+}))
+
+const ModalMenu = styled.div({
+  fontStyle: 'normal',
+  fontWeight: 500,
+  fontSize: '14px',
+  lineHeight: '21px',
+  color: '#7B7F93',
+  padding: '8px 0',
+  cursor: 'pointer',
+  ':hover': {
+    color: '#304FFE',
+  },
+})
+
+type IconImgProps = SpaceProps & {
+  isButton?: boolean
+}
+
+const IconImg = styled.img<IconImgProps>(
+  props => ({
+    width: '16px',
+    height: '16px',
+    cursor: props.isButton ? 'pointer' : 'auto',
+  }),
+  space
+)
+
+const HeaderColumn = styled(Flex)<FlexProps>(props => ({
+  justifyContent: 'center',
+  flexDirection: 'row',
+  alignItems: 'center',
+  flex: (props.flex as Property.Flex) || '3',
+}))
+
+const TableRow = styled(Flex)({
+  flexDirection: 'row',
+  alignItems: 'center',
+  minHeight: '50px',
+  borderTop: '1px dashed #DDDDE3',
+  justifyContent: 'space-evenly',
+})
+
+const TableColumn = styled(HeaderColumn)({})
+
+const TableContainer = styled(Flex)({
+  maxHeight: '200px',
+  flexDirection: 'column',
+  position: 'relative',
+})
+const TableHead = styled(Flex)({
+  flexDirection: 'row',
+  alignItems: 'center',
+  marginBottom: '8px',
+})
+
+const TableBody = styled(Flex)({
+  overflowY: 'scroll',
+  flexDirection: 'column',
+})
+
+interface ColumnDataProps {
+  title: string
+  colour?: string
+  flex?: string | number
+}
+
+interface BodyDataProps {
+  title: string
+  purchases: any[]
+  color?: string
+}
+
+interface TableProps {
+  headData: ColumnDataProps[]
+  bodyData: BodyDataProps[]
+  isClosed: boolean
+}
+
+// const body =  [ {type, }, amount, price ]
+
+export const Table = ({ headData, bodyData, isClosed }: TableProps) => {
+  const [tableMenu, setBidMenu] = useState<number>(-1)
+
+  const { isMobile } = useWindowSize()
+
+  const toggleBidMenu = (index: number) => {
+    if (tableMenu === index) {
+      setBidMenu(-1)
+      return
+    }
+    setBidMenu(index)
+  }
+  // <InfoImg src={InfoSVG} />
+  // <IconImg src={MoreSVG} marginRight="8px" isButton={true} onClick={() => toggleBidMenu(index)} />
+
+  return (
+    <TableContainer>
+      <TableHead>
+        {headData.map(({ title, flex }, index) => {
+          return (
+            <HeaderColumn key={index} flex={flex}>
+              <ColumnLabel>{title}</ColumnLabel>
+            </HeaderColumn>
+          )
+        })}
+      </TableHead>
+      <TableBody>
+        {bodyData.map(({ purchases, color, title }) =>
+          purchases.map((purchase: any, index) => {
+            return (
+              <TableRow key={index} padding={isMobile ? '0 8px' : '0 16px'}>
+                <TableColumn>
+                  <TokenPriceLabel color={color}>{title}</TokenPriceLabel>
+                </TableColumn>
+                {Object.getOwnPropertyNames(purchase).map((element: any, _index) => {
```

Comment:
I have only ever used keys() for this but I guess getOwnPropertyNames works too

Reply:
[It was brought up in a previous PR](https://github.com/cryptonative-ch/mesa-interface/pull/150#discussion_r630342546). If I want consistency then I shouldn't use keys().

Reply:
Nice, thanks for linking this. I didn't know about it before :)
I have always just used keys and never had an issue with it.